src/client/opamSwitchCommand.ml

```diff
@@ -262,21 +266,18 @@ let install_compiler
   else if OpamPackage.Set.is_empty has_comp_flag then
     OpamConsole.note
       "Packages %s don't have the 'compiler' flag set (nor any of their \
-       direct dependencies).\n\
+       dependencies).\n\
```

Comment:
I am not sure this message still makes sense if you allow any depth of dependencies: after all basically all packages will have a depend on a compiler at some depth, so it's unlikely to ever appear. The original intent was to avoid a warning for "aliases over compilers" packages like `ocaml`.

Reply:
Removed it then.